plaid/src/apis/github/environments.rs

```diff
@@ -0,0 +1,213 @@
+        let branch =
+            self.validate_branch_name(request.get("branch").ok_or(ApiError::BadRequest)?)?;
+
+        info!(
+            "Creating and configuring 'publish' environment in repo [{repo}] owned by [{owner}] on behalf of [{module}]"
+        );
+
+        // 1. Create the environment
+
+        let address = format!("/repos/{owner}/{repo}/environments/{ENVIRONMENT_NAME}");
+
+        let body = CreateEnvironmentPayload {
+            wait_timer: 0,
+            prevent_self_review: false,
+            reviewers: vec![],
+            deployment_branch_policy: DeploymentBranchPolicy {
+                protected_branches: false,
+                custom_branch_policies: true,
+            },
+        };
+
+        match self
+            .make_generic_put_request(address, Some(&body), module)
+            .await
+        {
+            Ok((status, _)) => {
+                if status == 200 {
+                    Ok(())
+                } else {
+                    Err(ApiError::GitHubError(GitHubError::UnexpectedStatusCode(
+                        status,
+                    )))
+                }
+            }
+            Err(e) => Err(e),
+        }?;
+
+        // 2. Create custom deployment protection rule to allow deployments only from the given branch
+        // See https://docs.github.com/en/rest/deployments/branch-policies?apiVersion=2022-11-28#create-a-deployment-branch-policy for more details
+
+        let address = format!(
+            "/repos/{owner}/{repo}/environments/{ENVIRONMENT_NAME}/deployment-branch-policies"
+        );
+
+        let body = CreateDeploymentBranchPolicyPayload {
+            name: branch.to_string(),
+            type_: "branch".to_string(), // it must be a branch, meaning it cannot be a tag that matches the given name
+        };
+
+        match self
+            .make_generic_post_request(address, Some(&body), module)
+            .await
+        {
+            Ok((status, _)) => {
+                if status == 200 {
+                    Ok(0)
+                } else {
+                    Err(ApiError::GitHubError(GitHubError::UnexpectedStatusCode(
+                        status,
+                    )))
+                }
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Configure a secret in a GitHub deployment environment
+    pub async fn configure_environment_secret(
+        &self,
+        params: &str,
+        module: &str,
+    ) -> Result<u32, ApiError> {
+        let request: HashMap<&str, &str> =
+            serde_json::from_str(params).map_err(|_| ApiError::BadRequest)?;
+
+        let owner = self.validate_username(request.get("owner").ok_or(ApiError::BadRequest)?)?;
+        let repo =
+            self.validate_repository_name(request.get("repo").ok_or(ApiError::BadRequest)?)?;
+        let environment_name = request.get("env_name").ok_or(ApiError::BadRequest)?;
```

Comment:
This creates a SSRF in the Plaid runtime. It's not sanitized so it gives almost total control over the URL on line `140`